File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.rs
impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32 {
      let mut hashset = HashSet::new();
      let mut longest = 0;


      for num in nums{
        hashset.insert(num);

      }
      
      for num_ref in &hashset{
        let num = *num_ref;
        let mut curr = 0;
        let mut i = 0;
        
        if hashset.contains(&(num - 1)){
            continue
        }else{
            while hashset.contains(&(num + i)){
                i += 1;
                curr += 1;

            }
        };

        longest = std::cmp::max(curr,longest);
      }


    longest



    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        counts = {}
        L = 0
        max_freq = 0
        longest = 0

        for R in range(len(s)):
            counts[s[R]] = counts.get(s[R], 0) + 1

            max_freq = max(max_freq, counts[s[R]])

            while (R - L + 1) - max_freq > k:
                counts[s[L]] -= 1
                L += 1
            
            longest = max(longest, R - L + 1)

        
        return longest